core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/LoggingAgentAutoRestartPatchService.java

```diff
@@ -0,0 +1,149 @@
+import com.sequenceiq.cloudbreak.core.bootstrap.service.ClusterBootstrapper;
+import com.sequenceiq.cloudbreak.core.bootstrap.service.ClusterDeletionBasedExitCriteriaModel;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.orchestrator.exception.CloudbreakOrchestratorFailedException;
+import com.sequenceiq.cloudbreak.orchestrator.host.TelemetryOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.cloudbreak.orchestrator.model.Node;
+import com.sequenceiq.cloudbreak.service.CloudbreakRuntimeException;
+import com.sequenceiq.cloudbreak.service.GatewayConfigService;
+import com.sequenceiq.cloudbreak.service.stack.InstanceMetaDataService;
+import com.sequenceiq.cloudbreak.service.stack.StackImageService;
+import com.sequenceiq.cloudbreak.util.CompressUtil;
+
+@Service
+public class LoggingAgentAutoRestartPatchService extends ExistingStackPatchService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingAgentAutoRestartPatchService.class);
+
+    private static final String AFFECTED_CDP_LOGGING_AGENT_VERSION = "0.2.13";
+
+    @Inject
+    private CompressUtil compressUtil;
+
+    @Inject
+    private TelemetryOrchestrator telemetryOrchestrator;
+
+    @Inject
+    private StackImageService stackImageService;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private ClusterComponentConfigProvider clusterComponentConfigProvider;
+
+    @Inject
+    private ClusterBootstrapper clusterBootstrapper;
+
+    @Override
+    public boolean isAffected(Stack stack) {
+        Version affectedVersion = Version.parse(AFFECTED_CDP_LOGGING_AGENT_VERSION);
+        try {
+            boolean affected = false;
+            if (StackType.WORKLOAD.equals(stack.getType())) {
+                Image image = stackImageService.getCurrentImage(stack);
+                Map<String, String> packageVersions = image.getPackageVersions();
+                if (packageVersions.containsKey(ImagePackageVersion.CDP_LOGGING_AGENT.getKey())
+                        && Version.parse(packageVersions.get(ImagePackageVersion.CDP_LOGGING_AGENT.getKey())).compareTo(affectedVersion) <= 0) {
+                    affected = true;
+                }
+            }
+            return affected;
+        } catch (Exception e) {
+            LOGGER.warn("Image not found for stack " + stack.getResourceCrn(), e);
+            throw new CloudbreakRuntimeException("Image not found for stack " + stack.getResourceCrn(), e);
+        }
+    }
+
+    @Override
+    void doApply(Stack stack) throws ExistingStackPatchApplyException {
+        if (isCmServerReachable(stack)) {
```

Comment:
Please keep in mind that this is not 100% reliable, improvement ticket was filed for it: [CB-14844](https://jira.cloudera.com/browse/CB-14844)